### ISSUE TITLE
Truncated Discrete Distributions

### ISSUE THREAD

**Issue by bdeonovic** [JuliaStats/Distributions.jl#373]
Does this package support truncated discrete distributions? I tried 

```julia
julia> Truncated(Geometric(1/4),1,5)
ERROR: `Truncated{D<:Distribution{Univariate,S<:ValueSupport}}` has no method matching Truncated{D<:Distribution{Univariate,S<:ValueSupport}}(::Geometric, ::Int64, ::Int64)
```

but continuous distributions work 
```julia
julia> Truncated(Normal(),1,5)
Truncated(Normal(μ=0.0, σ=1.0), range=(1.0, 5.0))
```

**Comment by simonbyrne** [comment]
Apparently not. But I wouldn't be opposed to making it more general.

**Comment by bdeonovic** [comment]
Taking a quick peek at the truncated code, would it be as simple as replacing `ContinuousUnivariateDistribution` with `UnivariateDistribution`

**Comment by simonbyrne** [comment]
It might be a little more complicated, as you will probably want an extra parameter for the support type so that `Truncated(Normal(),1,5) <: ContinuousUnivariateDistribution` and `Truncated(Poisson(),1,5) <: DiscreteUnivariateDistribution`

**Comment by bdeonovic** [comment]
I made a pull request to do this, could you let me know how it looks?https://github.com/JuliaStats/Distributions.jl/pull/375

**Comment by lindahua** [comment]
Seems addressed by #375.
